source/grid/grid_tools.cc

```diff
@@ -1714,8 +1715,33 @@ namespace GridTools
           }
         else
           {
-            closest_vertex_index =
-              GridTools::find_closest_vertex(mesh, p, marked_vertices);
+            if (used_vertices_rtree.size())
```

Comment:
Let us not use these confusing implicit conversions from numbers to `bool`.
```suggestion
            if (!used_vertices_rtree.empty())
```